garage/exploration_strategies/base.py

```diff
@@ -1,5 +1,8 @@
 class ExplorationStrategy:
-    def get_action(self, t, observation, policy, **kwargs):
+    def get_action(self, t, observation, policy, sess=None, **kwargs):
```

Comment:
why would i ever need to specify the session? This pattern isn't used anywhere else in primitives. 

Imagine how dangerous it could be if policy were in one session and you passed a different session to ExplorationStrategy.

I think you want something like

```python
with my_session:
    foo.get_action(t, obs, policy)
```

Reply:
since we are removing layers, sometimes policy needs to session to get the actions, e.g. in `discrete_qf_derived_policy`.
 
Do you mean we should use `with my_session:` in ExplorationStrategy and do `tf.get_default_session()` inside policy?

Reply:
You should not need a to initiate a session anywhere in your primitives because the algorithm should have done that for you.

i.e. `tf.get_default_session()` should always work inside your primitives.

Reply:
got it

Reply:
It looks like the change serves the purpose of removing `tensor_utils.compile_function()`. But if we are to replace it with `make_callable` instead of `sess.run()`, we may not need to feed sess here. The issue is beyond the scope of this PR. For this PR I suggest we do minimum or none API changes if possible. 
- Remove the API changes in exploration_strategy and sampler.
- Add get_qval in DiscreteMLPQFunction. This should avoid the need for passing sess to policy.

Reply:
As always, I defer to Chang's judgement on TF issues.